src/macaron/repo_finder/commit_finder.py

```diff
@@ -0,0 +1,480 @@
+            _, _, prefix = prefix.rpartition("/")
+        if prefix.lower() == artifact_name.lower():
+            named_tags.append(item)
+
+    if named_tags:
+        matched_tags = named_tags
+
+    # If multiple tags still remain, sort them based on the closest match in terms of individual parts.
+    if len(matched_tags) > 1:
+        matched_tags.sort(
+            key=lambda matched_tag: _count_parts_in_tag(matched_tag["version"], matched_tag["suffix"], parts)
+        )
+
+    return [_["tag"] for _ in matched_tags]
+
+
+def _count_parts_in_tag(tag_version: str, tag_suffix: str, version_parts: list[str]) -> int:
+    """Return a sort value based on how well the tag version and tag suffix match the parts of the actual version.
+
+    Parameters
+    ----------
+    tag_version: str
+        The tag's version.
+    tag_suffix: str
+        The tag's suffix.
+    version_parts: str
+        The version parts from the version string.
+
+    Returns
+    -------
+    int
+        The sort value based on the similarity between the tag and version, lower means more similar.
+
+    """
+    count = len(version_parts)
+    # Reduce count for each direct match between version parts and tag version.
+    tag_version_text = tag_version
+    for part in version_parts:
+        if part in tag_version_text:
+            tag_version_text = tag_version_text.replace(part, "", 1)
+            count = count - 1
+
+    # Try to reduce the count further based on the tag suffix.
+    if tag_suffix:
+        last_part = version_parts[-1]
+        # The tag suffix might consist of multiple version parts, e.g. RC1.RELEASE
+        suffix_split = split_pattern.split(tag_suffix)
+        if len(suffix_split) > 1:
+            # Try to match suffix parts to version.
+            versioned_string_match = False
+            for suffix_part in suffix_split:
+                if alphabetic_only_pattern.match(suffix_part) and suffix_part == last_part:
+                    # If the suffix part only contains alphabetic characters, reduce the count if it
+                    # matches the version.
+                    count = count - 1
+                    continue
+                if versioned_string.match(suffix_part):
+                    # If the suffix part contains alphabetic characters followed by numeric characters,
+                    # reduce the count if it matches the version (once only), otherwise increase the count.
+                    if not versioned_string_match and suffix_part == last_part:
+                        count = count - 1
+                        versioned_string_match = True
+                    else:
+                        count = count + 1
+
+        if tag_suffix != last_part:
+            count = count + 1
+        else:
+            count = count - 1
+
+    return count
+
+
+def _get_branch_of_commit(commit: Commit) -> str:
+    """Get the branch of the passed commit as a string or return None."""
+    branches = commit.branches
```

Comment:
It looks to that the [`Commit` ](https://gitpython.readthedocs.io/en/stable/reference.html#module-git.objects.commit)object doesn't have the `branches` attribute or method. The only object I think have the `branches` method is [`git.Repo`](https://gitpython.readthedocs.io/en/stable/reference.html#git.repo.base.Repo.branches) :thinking:

Reply:
`Commit` in this case is `pydriller.Commit` (which does have `branches`) , not `git.objects.Commit`. The overloaded name is a bit confusing.

Reply:
Righttt, my bad! It's indeed quite confusing.

Reply:
It's the `Commit` from `PyDriller` not `GitPytjpm`.